src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -544,6 +544,9 @@ protected SIPTransactionStack() {
         // The default (identity) address lookup scheme
 
         this.addressResolver = new DefaultAddressResolver();
+        
+        // Init vavles list
+        this.sipMessageValves = new ArrayList<SIPMessageValve>();
```

Comment:
yes, its part of the answer. ArrayList is a nonThreadSafe collection. Since we init this collection during Stack bootstrap, and then we just read from it, we dont need a ThreadSafe collection. 

In terms of size, and how the collection will grow as new elements are added, ArrayList starts with a 10 elements capacity. Ideally the collection should be started to the same number of cofigured valves. Again, since this collection will be only written during bootstrap, its not really important.

So,yes, ArrayList is the best matching collection here. I need you to think about threadSafe, collectionGrowing/Shrinking, and the like, everytime you select a collection.